File: PHP/src/C/Checkout.php
<?php

namespace Org\Codecop\Dependencies\C;

class Checkout {
    public function createReceipt(Money $amount): Receipt {
        $receipt = new Receipt();
        $vat = $amount->percentage(20);

        $receipt->setAmount($amount);
        $receipt->setTax($vat);
        $receipt->setTotal($amount->add($vat));

        ReceiptRepository::store($receipt);

        return $receipt;
    }
}

File: Java/src/main/java/org/codecop/dependencies/c/Checkout.java
package org.codecop.dependencies.c;

public class Checkout {

    public Receipt createReceipt(Money amount) {
        Receipt receipt = new Receipt();
        Money vat = amount.percentage(20);

        receipt.setAmount(amount);
        receipt.setTax(vat);
        receipt.setTotal(amount.add(vat));

        store(receipt);

        return receipt;
    }

    protected void store(Receipt receipt) {
        ReceiptRepository.store(receipt);
    }
}
